### PR TITLE
[JEP-16] - Inclusive terminology guidelines and continuous updates

### PR DESCRIPTION
**Pull request by aHenryJard**
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

I think we need a place to have the official, community validated list of words and replacement. I propose to do it in a JEP to have git history.

With possible link in documentation here https://www.jenkins.io/doc/developer/internationalization/

### CODE REVIEW

File: jep/0000/README.adoc [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+[cols="1h,1"]
+|===
+| JEP
+| 0000
+
+| Title
+| Continuous terminology update and identified offensive words including translations
+
+| Sponsor
+| :bulb: link:https://github.com/aHenryJard[aHenryJard], link:https://github.com/username2[User Name 2] :bulb:
```
Comment (oleg-nenashev) [inline jep/0000/README.adoc:43]: Username 2? :)

File: jep/0000/README.adoc [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+====
+
+.Metadata
+[cols="1h,1"]
+|===
+| JEP
+| 0000
+
+| Title
+| Continuous terminology update and identified offensive words including translations
```
Comment (oleg-nenashev) [inline jep/0000/README.adoc:40]: It us a bit wordy IMHO. I am also not 100% comfortable about "offensive". Nit all non inclusive terminology is considered offensive by the community. I would rather prefer to align with https://inclusivenaming.org/ guidelines and standards. How about this one?

```suggestion
| Inclusive terminology guidelines and continuous updates
```

File: jep/0000/README.adoc [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+| Created
+| Date (2021-06-22)
+
+| BDFL-Delegate
+| TBD
+
+
+| JIRA
+| https://issues.jenkins.io/browse/JENKINS-65398[JENKINS-65398 renaming Master to Controller/Built-in node ; whitelist and blacklist deprecation]
```
Comment (oleg-nenashev) [inline jep/0000/README.adoc:60]: Reference agent cleanup as well? Maybe also the GitHub project?

File: jep/0000/README.adoc [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+| slave                          | agent                         | agents (plural)                                  |
+| master                         | controller                    | about the Jenkins instance                       |
+| master                         | built-in node                 | about the node embedded in the Jenkins instance  |
+| whitelist                      | allowlist                     | default but can be adapted to context            |
+| blacklist                      | denylist or blocklist         | default but can be adapted to context            |
+|======================
+
+=== Translations
+
+==== French - Français
```
Comment (oleg-nenashev) [inline jep/0000/README.adoc:95]: Could you please move all languages? Or do yio want to get format reviews first?

File: jep/0000/README.adoc [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+| allowlist                     |                               |                               |                                                            |
+| denylist or blocklist         |                               |                               |                                                            |
+|======================
+
+* Known Exceptions
+** esclave is valid in https://github.com/jenkinsci/jenkins/blob/677fbb2a81fa8e9a0bc37ad5eca56565612e1f2e/core/report-l10n.rb#L173
+
+== Motivation
+
+There is more and more deprecated words and replacement. Discussions and motivations on new term can be done in discussion channels, but once the descision is made by the community we need to have all deprecated terminology and replacement at one place.
```
Comment (oleg-nenashev) [inline jep/0000/README.adoc:116]: Protip: split sentences to lines for easier editing and reviews

File: jep/0000/README.adoc (hunk 1 above)
Comment (aHenryJard) [reply jep/0000/README.adoc:43]: yes I am hoping to not be alone :) I wait a bit to keep the syntax

File: jep/0000/README.adoc (hunk 3 above)
Comment (aHenryJard) [reply jep/0000/README.adoc:60]: I did it first, but I think you can do only one link in the metadata table, so I have move them to the reference part

File: jep/0000/README.adoc (hunk 4 above)
Comment (aHenryJard) [reply jep/0000/README.adoc]: done ;-) yes I was doing the formatting part with French only to have an easy copy/paste after

**Comment by daniel-beck** [comment]
Seems kind of weird to have this now as a "documentation JEP" when the actual decisionmaking went without.

Would this better off elsewhere? I would expect expected terms for various parts of Jenkins in French to need documenting anyway, elsewhere (we used to use wiki pages for that). For English terms, we have https://www.jenkins.io/doc/book/glossary/

**Comment by aHenryJard** [comment]
> Seems kind of weird to have this now as a "documentation JEP" when the actual decisionmaking went without.
> 
> Would this better off elsewhere? I would expect expected terms for various parts of Jenkins in French to need documenting anyway, elsewhere (we used to use wiki pages for that). For English terms, we have https://www.jenkins.io/doc/book/glossary/

Well I think that maybe it's not the last term to be changed, and the decision making is not fully done yet, at least for languages there is some missing terms. I hesitated with documentation, I guess it's easy to move it there if the JEP is not the right format. (I hope it's also adoc in documentation now :) )

File: jep/0000/README.adoc [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+|===
+| JEP
+| 0000
+
+| Title
+| Inclusive terminology guidelines and continuous updates
+
+| Sponsor
+| link:https://github.com/aHenryJard[aHenryJard]
+
```
Comment (mikecirioli) [inline jep/0000/README.adoc:44]: feel free to add `| link:https://githumb.com/mikecirioli[Mike Cirioli]`

File: jep/0000/README.adoc [hunk 7]
```diff
@@ -0,0 +1,338 @@
...
+The goal of this JEP is to have the latest status of deprecated word or terms and agreed replacements in a table at one place that can be discussed with history.
+The proposal is to have git to keep history of changes, and discussion on new term on PRs.
+
+Deprecated term are written explicitly to be searchable.
+
+== Specification
+
+* The term in table must be in lower case (when the language allows it). 
+
```
Comment (oleg-nenashev) [inline jep/0000/README.adoc:78]:
```suggestion
```

File: jep/0000/README.adoc [hunk 8]
```diff
@@ -0,0 +1,338 @@
...
+* To add a new language, propose a PR on top of this JEP by a copy/paste one language (other than English) and add a new section.
+
```
Comment (oleg-nenashev) [inline jep/0000/README.adoc:80]: It is not a list otherwise 

```suggestion
```

File: jep/0000/README.adoc [hunk 9]
```diff
@@ -0,0 +1,338 @@
...
+|======================
+
+=== Translations
+
+==== French
+
+* property suffix: fr
+
+* Table
+
```
Comment (oleg-nenashev) [inline jep/0000/README.adoc:106]: Do you really need bullets here? 

```suggestion
==== French (*-fr.properties)

```

File: jep/0000/README.adoc [hunk 10]
```diff
@@ -0,0 +1,338 @@
...
+| English term                  |Deprecated term                | New term root                 | Comments                                                 
+| agent                         | esclave, esclaves             | agent                         | "esclave" can be valid in a context of a spoken language.
+| controller                    | maître, maitre                | contrôleur                    |                                                          
+| built-in node                 | maître, maitre                | TDB                           |                                                          
+| allowlist                     |                               |                               |                                                          
+| denylist or blocklist         |                               |                               |                                                          
+|======================
+
+* Known Exceptions
+** esclave is valid in https://github.com/jenkinsci/jenkins/blob/677fbb2a81fa8e9a0bc37ad5eca56565612e1f2e/core/report-l10n.rb#L173
```
Comment (oleg-nenashev) [inline jep/0000/README.adoc:118]: Not sure the developer tooling is a concern TBH. Better to add a comment to the code than to keep such a minor exception in the JEP

```suggestion
Known Exceptions
* `esclave` is valid in https://github.com/jenkinsci/jenkins/blob/677fbb2a81fa8e9a0bc37ad5eca56565612e1f2e/core/report-l10n.rb#L173
```

File: jep/0000/README.adoc [hunk 11]
```diff
@@ -0,0 +1,338 @@
...
+| built-in node                 | TBD                           | TDB                           |               
+| allowlist                     |                               |                               |               
+| denylist or blocklist         |                               |                               |               
+|======================
+
+* Known Exceptions
+
+== Motivation
+
+There is more and more deprecated words and replacement.
```
Comment (oleg-nenashev) [inline jep/0000/README.adoc:266]:
```suggestion
There is more and more deprecated terms and replacements.
```

File: jep/0000/README.adoc [hunk 12]
```diff
@@ -0,0 +1,338 @@
...
+* Table
+
+[options="header"]
+|======================
+| English term                  |Deprecated term                | New term root                 | Comments                                                 
+| agent                         | esclave, esclaves             | agent                         | "esclave" can be valid in a context of a spoken language.
+| controller                    | maître, maitre                | contrôleur                    |                                                          
+| built-in node                 | maître, maitre                | TDB                           |                                                          
+| allowlist                     |                               |                               |                                                          
+| denylist or blocklist         |                               |                               |
```
Comment (oleg-nenashev) [inline jep/0000/README.adoc:114]: IIRC the consensus was to avoid `blocklist` in our codebase and terms. CC @halkeye @uhafner to confirm

```suggestion
| denylist         |                               |                               |                                                          
```

File: jep/0000/README.adoc [hunk 13]
```diff
@@ -0,0 +1,338 @@
...
+If some or all of the changes will require human interaction to verify them, explain why automated tests are considered impractical.
+Then, summarize what kinds of test cases might be required: user scenarios with action steps and expected outcomes.
+Detail whether behavior might be different based on the platform (operating system, servlet container, web browser, etc.)?
+Are there foreseeable interactions between different permissible versions of components (Jenkins core, plugins, etc.)?
+Does this change require that any special tools, proprietary software, or online service accounts to exercise a related code path (e.g., Active Directory server, GitHub login, etc.)?
+When will you complete testing relative to merging code changes, and might retesting be required if other changes are made to this area in the future?
+
+If this proposal requires no testing, this section may simply say:
+There are no testing issues related to this proposal.
+====
```
Comment (oleg-nenashev) [inline jep/0000/README.adoc:330]:
```suggestion
N/A
```

File: jep/0000/README.adoc [hunk 14]
```diff
@@ -0,0 +1,338 @@
...
+
+[TIP]
+====
+Describe any incompatibilities and their severity.
+Describe how the JEP proposes to deal with these incompatibilities.
+
+If there are no backwards compatibility concerns, this section may simply say:
+There are no backwards compatibility concerns related to this proposal.
+====
+
```
Comment (oleg-nenashev) [inline jep/0000/README.adoc:295]:
```suggestion
```

File: jep/0000/README.adoc [hunk 15]
```diff
@@ -0,0 +1,338 @@
...
+[cols="1h,1"]
+|===
+| JEP
+| 0000
+
+| Title
+| Inclusive terminology guidelines and continuous updates
+
+| Sponsor
+| link:https://github.com/aHenryJard[aHenryJard]
```
Comment (oleg-nenashev) [inline jep/0000/README.adoc:43]: I would recommend adding a display name here unless you have a strong preference

File: jep/0000/README.adoc [hunk 16]
```diff
@@ -0,0 +1,306 @@
...
+====
+
+[TIP]
+====
+*BDFL-Delegate* is uncommented by default.
+As part of the in initial conversation or the JEP submission the sponsor should
+look for a BDFL Delegate.
+While not required, it is better for the community if Delegates perform most reviews.
+If no suitable BDFL-Delegate can be found, that row may be commented out.
+====
```
Comment (oleg-nenashev) [inline jep/0000/README.adoc:31]:
```suggestion
```

**Comment by oleg-nenashev** [comment]
> Seems kind of weird to have this now as a "documentation JEP" when the actual decisionmaking went without.

Not the first time, does not seem to be a problem. Actually it happens all the time for process JEPs. When someone starts a process JEP, there is usually consensus built in the mailing list first. There are also some security related JEPs which also happen without preliminary discussion outside the security team.

> Would this better off elsewhere?

I have no problem with putting it into JEP. Although I do not feel strongly it **must** be a JEP (Google Doc is okay with me), having it as JEP actually raises the profile of the terminology updates topic. Looks perfectly fine, with assumption that this process JEP will keep growing and evolving along with the community

File: jep/0000/README.adoc (hunk 15 above)
Comment (aHenryJard) [reply jep/0000/README.adoc:43]: oki :)

**Review by oleg-nenashev — APPROVED** [review]
I suggest patching it incrementally. Current draft definitely matches minimum requirements for the draft JEP
